src/main/java/rx/internal/operators/OperatorOnBackpressureLatest.java

```diff
@@ -0,0 +1,225 @@
+    /**
+     * A terminatable producer which emits the latest items on request.
+     * @param <T>
+     */
+    static final class LatestEmitter<T> extends AtomicLong implements Producer, Subscription, Observer<T> {
+        /** */
+        private static final long serialVersionUID = -1364393685005146274L;
+        final Subscriber<? super T> child;
+        LatestSubscriber<? super T> parent;
+        final AtomicReference<Object> value;
+        /** Written before done, read after done. */
+        Throwable terminal;
+        volatile boolean done;
+        /** Guarded by this. */
+        boolean emitting;
+        /** Guarded by this. */
+        boolean missed;
+        static final Object EMPTY = new Object();
+        static final long NOT_REQUESTED = Long.MIN_VALUE / 2;
+        public LatestEmitter(Subscriber<? super T> child) {
+            this.child = child;
+            this.value = new AtomicReference<Object>(EMPTY);
+            this.lazySet(NOT_REQUESTED); // not 
+        }
+        @Override
+        public void request(long n) {
+            if (n >= 0) {
+                for (;;) {
+                    long r = get();
+                    if (r == Long.MIN_VALUE) {
+                        return;
+                    }
+                    long u;
+                    if (r == NOT_REQUESTED) {
+                        u = n;
+                    } else {
+                        u = r + n;
+                        if (u < 0) {
+                            u = Long.MAX_VALUE;
+                        }
+                    }
+                    if (compareAndSet(r, u)) {
+                        if (r == NOT_REQUESTED) {
+                            parent.requestMore(Long.MAX_VALUE);
+                        }
+                        emit();
+                        return;
+                    }
+                }
+            }
+        }
+        long produced(long n) {
+            for (;;) {
+                long r = get();
+                if (r < 0) {
+                    return r;
+                }
+                long u = r - n;
+                if (compareAndSet(r, u)) {
+                    return u;
+                }
+            }
+        }
+        @Override
+        public boolean isUnsubscribed() {
+            return get() == Long.MIN_VALUE;
+        }
+        @Override
+        public void unsubscribe() {
+            if (get() >= 0) {
+                getAndSet(Long.MIN_VALUE);
+            }
+        }
+        
+        @Override
+        public void onNext(T t) {
+            value.lazySet(t); // emit's synchronized block does a full release
+            emit();
+        }
+        @Override
+        public void onError(Throwable e) {
+            terminal = e;
+            done = true;
+            emit();
+        }
+        @Override
+        public void onCompleted() {
+            done = true;
+            emit();
+        }
+        void emit() {
+            synchronized (this) {
```

Comment:
> any point in optimizing the emission further and I think a smaller overhead would just drop more values.

The scenario I'm considering is where this operator is put in place to handle occasional bursts where generally the consumer is fast enough, but occasionally it is slow. This is a common use of `onBackpressureDrop` for us in our stream processor. We would use this in the same way.